### ISSUE TITLE
undefined bin makes api crash

### ISSUE THREAD

**Issue by fdnieves** [aerospike/aerospike-client-nodejs#23]
Hello, the following code crashes the API and makes a stack dump .

	var key = aerospike.key('test','test',1);
	var doc = {'bla': 1}
	doc.asd = undefined;
	
	client.put(key, doc, function(err) 
	{
	   //do stuff
	});

We cannot handle this error, as this is not being caught by process.on('uncaughtException') or domains. Is this a bug? Is there any way we can handle this kind of errors?

The error is something like this:

#
# Fatal error in ../deps/v8/src/api.h, line 297
# CHECK(allow_empty_handle || that != __null) failed
#

==== Stack trace ============================================

Security context: 0x2ffb4aa06b71 <JS Object>#0#
.....

Thanks,

**Comment by pablonazareno** [comment]
I'm having the same issue.
I'm using verion 1.0.25. 

Thanks

**Comment by GayathriKaliyamoorthy** [comment]
Hi 
The latest release has the fix and we are not able to reproduce this inhouse. Could you give us the code snippet for us to dig deeper.

Thanks.

**Comment by pablonazareno** [comment]
Hi
The process don't crush actually, but have some strange behaviour in stored data.

Try this code:
```javascript
var aerospike = require('aerospike');
var client = aerospike.client(connectionSetting);
client.connect(function(response) {
    if (response.code === aerospike.status.AEROSPIKE_OK) {
        var digest = aerospike.key('test', 'test', 'title');
        var value = {
            header: {
                banners: [],
                apply_banner: "New Year!",
                empty: undefined
            }
        };
        client.put(digest, value, console.log);
    } else {
        console.error('SOMETHING IS WRONG');
    }
});
```
Generate this output:
```
Jan 07 2015 14:27:29 GMT: ERROR(5679) [conversions.cc:515] [asval_from_jsobject] - Object passed is undefined
{ code: 0, message: null, func: null, file: null, line: null } { ns: 'test', set: 'test', key: 'title' }
```
And at Aerospike DB:
```
ascli get test test title
{"header": { }}
```
but Using AQL
```
gt;aql
Aerospike Query
Copyright 2013 Aerospike. All rights reserved.

aql> select * from test.test
Segmentation fault (core dumped)
```
Thanks

Pablo

**Comment by iamontheinet** [comment]
Hi Pablo,

The issue has been resolved as of 1.0.30.

Here's is the output from running your code snippet:

ASCLI: `ascli get test test title`
Result: `{"header": { "banners": [ ], "empty": null, "apply_banner": "New Year!" }}`

AQL: `select * from test.test`
Result: `{"banners":[], "empty":NIL, "apply_banner":"New Year!"} `

Hope this helps.

Dash

**Comment by GayathriKaliyamoorthy** [comment]
@pablonazareno  Could you confirm if this works for you, so that we can close this issue?

Thanks.
